### DISCUSSION TITLE
Getting “Initializing project…” repeatedly in LazyVim (monorepo setup)

### DISCUSSION THREAD

**Discussion by dan-nil** [LazyVim/LazyVim#5086] [Q&A]
In the bottom right corner of the screen, I repeatedly see this message:

Initializing project...

It eventually finishes with one of the tsconfig.json files, but then it starts over with the same tsconfig.json

![image](https://github.com/user-attachments/assets/f9369cec-9964-4b35-a3d5-9b21032cb47b)

Does anyone know why this might be happening?

Thanks in advance!

**Comment by folke** [comment]
Make sure to update lspconfig, since there was an issue there over the last couple of days where the root dir was no longer detected

**Reply by dan-nil** [reply]
> Make sure to update lspconfig, since there was an issue there over the last couple of days where the root dir was no longer detected

I updated lspconfig, but I still have the same issue.
After a few minutes, I am getting this one:
<img width="605" alt="image" src="https://github.com/user-attachments/assets/dd784a6e-a87e-496c-ba20-70ebd1ea1a3e" />

**Reply by dan-nil** [reply]
I was able to fix the issue by switching from `vtsls` to `ts_ls`, but I couldn’t figure out the root cause.

**Reply by Raagh** [reply]
I am experiencing the same, it started happening a few days go. It is constantly crashing

**Reply by dpetka2001** [reply]
https://github.com/yioneko/vtsls?tab=readme-ov-file#server-stuck-or-crashed-on-large-repo Maybe related to this? It seems it's known in `vtsls` repo and has the steps above to potentially troubleshoot it.

**Reply by Raagh** [reply]
I can confirm setting this seems to have solved the issue, although I am not sure why it started happening only a few days ago. I have been coding in the same monorepo for over a year

```lua
  {
    "neovim/nvim-lspconfig",
    opts = function(_, opts)
      opts.servers = {
        vtsls = {
          settings = {
            typescript = {
              tsserver = {
                maxTsServerMemory = 8192,
              },
            },
          },
        },
      }
  },


```

**Reply by dan-nil** [reply]
> I can confirm setting this seems to have solved the issue, although I am not sure why it started happening only a few days ago. I have been coding in the same monorepo for over a year

```lua
  {
     "neovim/nvim-lspconfig",
     opts = function(_, opts)
       opts.servers = {
         vtsls = {
           settings = {
             typescript = {
               tsserver = {
                 maxTsServerMemory = 8192,
               },
             },
           },
         },
       }
     end
   },
 ```
 
 I corrected your block of code ✌🏻

I will give it a try. Thank!
